projects/igniteui-angular/src/lib/query-builder/query-builder-tree.component.html

```diff
@@ -21,40 +21,58 @@
             </igx-select-item>
         </igx-select>
 
-        <igx-combo
-            #selectedReturnFieldsCombo
-            [itemsMaxHeight]="250"
-            [data]="fields"
-            [displayKey]="'field'"
-            [valueKey]="'field'"
-            [disabled]="!selectedEntity"
-            [(ngModel)]="selectedReturnFields"
-            [overlaySettings]="returnFieldSelectOverlaySettings"
-            searchPlaceholder="{{ this.resourceStrings.igx_query_builder_search }}"
-            [style.display]="isInEditMode() ? 'block' : 'none'"
-            (selectionChanging)="onReturnFieldSelectChanging($event)"
-        >
-            <ng-template igxComboHeader>
-                <div
-                    class="igx-drop-down__item igx-drop-down__item--query-builder"
-                    (click)="onSelectAllClicked($event)"                
-                    [ngClass]="{'igx-drop-down__item--selected': this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields.length === this.selectedReturnFields.length}"
-                >
-                    <igx-checkbox
-                            [checked]="this.selectedEntity && this.selectedReturnFields && this.selectedEntity.fields.length === this.selectedReturnFields.length"
-                            [indeterminate]="this.selectedEntity && this.selectedReturnFields && this.selectedReturnFields.length > 0 && this.selectedReturnFields.length < this.selectedEntity.fields?.length"
-                            [readonly]="true"
-                            [disableRipple]="true"
-                            [tabindex]="-1"
-                            class="igx-combo__checkbox">
-                    </igx-checkbox>
-                    <div class="igx-drop-down__inner">
-                        {{ this.resourceStrings.igx_query_builder_select_all }}
+        <ng-container *ngIf="!parentExpression">
```

Comment:
Instead of ng-containers with ngIf, you can use 
@if(parentExpression) {
            <igx-select #returnFieldSelect
                ...
            </igx-select>
           } @else {
               <igx-combo
                   #returnFieldsCombo
                   ...
               </igx-combo>
           }